libp2p/multibase.nim

```diff
@@ -439,48 +442,50 @@ proc decode*(mbtype: typedesc[MultiBase], inbytes: openarray[char],
     result = mb.decr(inbytes.toOpenArray(1, length - 1), outbytes, outlen)
 
 proc encode*(mbtype: typedesc[MultiBase], encoding: string,
-             inbytes: openarray[byte]): string =
+             inbytes: openarray[byte]): Result[string, string] =
```

Comment:
Any reason here to not use `Result[string, cstring]` ?

Reply:
the string has runtime dynamic text

Reply:
but yeah we could remove the dynamic part

Reply:
Ah, right, didn't see the third `err`